<?xml version="1.0"?>
<!-- ROS head file to run the 7th floor rosbag and rviz   -->
<launch>
  <!-- robot configuration parameters -->
  <arg name="robot" default="vizzy"/> <!-- full, lower_body, foot -->

  <arg name="map_frame" default="map" />
  <arg name="tf_prefix" default="" />
  <arg name="base_frame_id" default="$(arg tf_prefix)/base_footprint" />
  <arg name="odom_frame_id" default="$(arg tf_prefix)/odom" />
  <arg name="map_topic" default="/map" />
  <arg name="visualization" default="true" />

<!-- Run pedestrian detector !-->
  <arg name="image_in" default="/vizzy/l_camera/image_rect_color" />
  <arg name="image_out" default="pedestrian_detector_image" />
  <node pkg="pedestrian_detector" type="detector" name="detector" output="screen">
  		<param name="image_transport" value="compressed" />
		<remap from="image_in" to="$(arg image_in)"/>
		<remap from="image_out" to="$(arg image_out)"/>
  </node>

<!-- Run the tracker !-->
 <node pkg="pedestrian_detector" type="tracker" name="tracker" output="screen">
  </node>


<!-- Run the follower !-->
 <node pkg="pedestrian_detector" type="follower" name="follower" output="screen">
  </node>


  <!-- Play the bag file. Leave it in a loop !-->

<node pkg="rosbag" type="play" name="player" output="screen" args="--clock -l $(find pedestrian_detector)/bags/first.bag"/>

      <!-- move_base using local eband planner -->
      <include file="$(find vizzy_navigation)/launch/move_base_eband_local.launch">
        <arg name="map_topic" value="$(arg map_topic)" />
      </include>

  <!-- Load the URDF into the ROS Parameter Server -->
  <param name="robot_description" command="$(find xacro)/xacro.py $(find vizzy_description)/robots/vizzy.urdf.xacro use_yarp:=false" />

  <!-- launch vizualization on rviz -->
  <group if="$(arg visualization)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d vizzy.rviz" />
  </group>

</launch>
